Reading Page + Cooking Page

AsianHam opened the pull request.

AsianHam committed:
name update

updated interface name

AsianHam committed:
reading page

Setup routes

Created Reading Page layout

Set thumbnails to be cropped on home page

Updated interface to include pictures on every step

CSS cleanup

AsianHam committed:
image crop

set image to be cropped on the reading page

AsianHam committed:
Cooking Page

Created Cooking Page layout and associated CSS

Created required routes to Cooking Page

updates urls to use environment variables

Jacksondr5 commented on src/CookingPages.tsx:
```diff
@@ -0,0 +1,167 @@
+  };
+
+  useEffect(() => {
+    setLoading(true);
```
I dont think you need to do this since its default set to true

Jacksondr5 commented on src/CookingPages.tsx:
```diff
@@ -0,0 +1,167 @@
+    getRecipe();
+  }, []);
+
+  const renderHeader = () => {
```
You've got a lot of redundant code here.  Can you put the if logic inside the divs?  IE, inside the step bar, have an if for the previous button and an if for the next button

Jacksondr5 commented on src/CookingPages.tsx:
```diff
@@ -0,0 +1,167 @@
+    }
+  };
+
+  const renderImage = () => {
```
This doesn't need to be a function, you can assign JSX to a variable and just reference that in the return

Jacksondr5 commented on src/ReadingPages.tsx:
```diff
@@ -0,0 +1,122 @@
+  const [readingRecipe, setReadingRecipe] = useState<Recipe>(emptyRecipe);
+  const [loading, setLoading] = useState(true);
+
+  const id = useParams().recipeId;
```
I would move this fetching logic to a separate file, that way you dont need to have it 3 times

Jacksondr5 commented on src/ReadingPages.tsx:
```diff
@@ -0,0 +1,122 @@
+  };
+
+  useEffect(() => {
+    setLoading(true);
```
Also dont need to do this

Jacksondr5 commented on src/ReadingPages.tsx:
```diff
@@ -0,0 +1,122 @@
+  };
+
+  const showImage = (imageSrc: string) => {
+    if (imageSrc !== "") {
```
You have image marked as a non nullable string in your types.  Do you need to make it a nullable string?

AsianHam committed:
pr changes

created separate files to handle repetitive logic

shortened redundant code into individual functions

general cleanup

AsianHam committed:
pr update

forgot to remove commented out code.

AsianHam committed:
pr changes

removed functions for conditional formatting

moved fetch calls to another file

added conditional formatting inside buttons
